File: backend/src/main/scala/kpn/api/common/node/NodeIntegrityDetail.scala
package kpn.api.common.node

import kpn.api.common.RouteScope
import kpn.api.common.RouteType
import kpn.api.common.common.Ref
import kpn.api.custom.ScopedRouteType

case class NodeIntegrityDetail(
  routeType: RouteType,
  routeScope: RouteScope,
  expectedRouteCount: Int,
  routeRefs: Seq[Ref]
) {

  def failed: Boolean = {
    routeRefs.sizeIs != expectedRouteCount
  }

  def hasScopedRouteType(scopedRouteType: ScopedRouteType): Boolean = {
    scopedRouteType.routeType == routeType && scopedRouteType.routeScope == routeScope
  }
}

File: backend/api/src/main/java/kpn/api/common/node/NodeIntegrityDetail.java
package kpn.api.common.node;

import kpn.api.common.RouteScope;
import kpn.api.common.RouteType;
import kpn.api.common.common.Ref;

import com.google.common.collect.ImmutableList;

public record NodeIntegrityDetail(
  RouteType routeType,
  RouteScope routeScope,
  Integer expectedRouteCount,
  ImmutableList<Ref> routeRefs
) {}

/* TODO migrate

  def failed: Boolean = {
    routeRefs.sizeIs != expectedRouteCount
  }

  def hasScopedRouteType(scopedRouteType: ScopedRouteType): Boolean = {
    scopedRouteType.routeType == routeType && scopedRouteType.routeScope == routeScope
  }

*/
